backend/app.py

```
@@ -23,12 +23,16 @@
-    app = Flask(__name__)
+    app = Flask(__name__, instance_relative_config=True)
+
+    # Create instance folder if it doesn't exist
+    os.makedirs(app.instance_path, exist_ok=True)
 
     # Configure the app with necessary settings
     app.config["SECRET_KEY"] = os.environ.get("SECRET_KEY", "default_secret_key")
     app.config["SQLALCHEMY_DATABASE_URI"] = os.environ.get(
-        "SQLALCHEMY_DATABASE_URI", "sqlite:///site.db"
+        "SQLALCHEMY_DATABASE_URI", 
+        f"sqlite:///{os.path.join(app.instance_path, 'site.db')}"
```

Ruff (0.8.2)

Trailing comma missing

Add trailing comma

(COM812)